discovery/dns/dns.go

```diff
@@ -136,27 +136,28 @@ func (d *Discovery) refresh(ctx context.Context, name string, ch chan<- []*confi
 		return model.LabelValue(net.JoinHostPort(a, fmt.Sprintf("%d", p)))
 	}
 
-	for _, record := range response.Answer {
-		target := model.LabelValue("")
-		switch addr := record.(type) {
-		case *dns.SRV:
-			// Remove the final dot from rooted DNS names to make them look more usual.
-			addr.Target = strings.TrimRight(addr.Target, ".")
-
-			target = hostPort(addr.Target, int(addr.Port))
-		case *dns.A:
-			target = hostPort(addr.A.String(), d.port)
-		case *dns.AAAA:
-			target = hostPort(addr.AAAA.String(), d.port)
-		default:
-			d.logger.Warnf("%q is not a valid SRV record", record)
-			continue
+	if response != nil {
```

Comment:
With the change below to not return nil, this check and the additional indentation becomes unnecessary and can be reverted again.